libs/sdk-core/src/breez_services.rs

```diff
@@ -294,6 +294,8 @@ impl BreezServices {
         {
             Some(_) => Err(SendPaymentError::AlreadyPaid),
             None => {
+                self.validate_liquidity(amount_msat)?;
```

Comment:
We don't force sync before paying so the channels balance might not be synced as well with the node backend.
I think the reasonable way to solve it and return a proper timely mannered response is by adding this optimization to c-lightning.

Reply:
I raised an issue in Greenlight last week related to this: https://github.com/Blockstream/greenlight/issues/361

Reply:
I agree, the right place to solve it is GL.

I'll close this, as it will be addressed by the GL issue above.